source/common/grpc/async_client_manager_impl.cc

```diff
@@ -0,0 +1,44 @@
+#include "common/grpc/async_client_manager_impl.h"
+
+#include "common/grpc/async_client_impl.h"
+
+namespace Envoy {
+namespace Grpc {
+
+AsyncClientFactoryImpl::AsyncClientFactoryImpl(Upstream::ClusterManager& cm,
+                                               const std::string& cluster_name)
+    : cm_(cm), cluster_name_(cluster_name) {
+  auto clusters = cm_.clusters();
```

Comment:
Does using `get()` here like was done in the access log code not work because TLS clusters are not setup yet in some cases?

Reply:
That's right; there's a chick-and-egg situation with `ClusterManager` initialization, where EDS based clusters are created, and `AsyncClientManager` initialization which takes place in that same constructor.

Reply:
OK sounds good.